cmake/Modules/FindCatch_EP.cmake

```diff
@@ -73,8 +73,8 @@ if (NOT Catch2_FOUND AND TILEDB_SUPERBUILD)
     PREFIX "externals"
     # Set download name to avoid collisions with only the version number in the filename
     DOWNLOAD_NAME ep_catch.zip
-    URL "https://github.com/catchorg/Catch2/archive/v3.1.0.zip"
-    URL_HASH SHA1=b23753594a743feabd4e30f83b31ebb31081092a
+    URL "https://github.com/catchorg/Catch2/archive/v3.3.2.zip"
```

Comment:
How about you pin the Catch version in https://github.com/TileDB-Inc/TileDB/blob/bc4638540708f4c3a1148c34d77f95b9c33ce193/vcpkg.json#L76 ?

Reply:
There is no connection between the ExternalProject_Add and vcpkg.json

Reply:
(in theory we could [parse it in cmake](https://cmake.org/cmake/help/latest/command/string.html#id2), but *let's not*)

Reply:
I meant you do update `vcpkg.json` in addition to the EP. There is no connection but it would ensure we are using the same Catch version regardless of whether vcpkg is enabled.